openmc/mesh.py

```diff
@@ -786,6 +792,16 @@ def write_data_to_vtk(self, filename, datasets, volume_normalization=True):
             volume_normalization=volume_normalization
         )
 
+    def is_flat(self):
+        """Returns True if the mesh is flat
+        """
```

Comment:
```suggestion
        """Checks if the mesh is flat in one or more dimensions. Returns True
        if the mesh is flat and False if the mesh is 3D.

        Returns
        -------
        flatness : bool
            whether or not the mesh is flat

        """
```

I think the doc string might want a return type. I randomly added some content but main comment is could we have a return type for the doc string

Reply:
Do we think an `is_flat` method is generally useful in the API or do we just want to prevent people from creating a zero-thickness mesh in the first place? I'd suggest a few additional checks in the property setters instead. i.e making sure `dimension` and `width` are always above 0 and in the `upper_right` and `lower_left` setters just check if the other has been set and if so don't allow the same value to be set.

Reply:
@eepeterson you're right we want to avoid creating zero-thickness meshes. This method is_flat is only called in the `upper_right` and `lower_left` setters.

I can get rid of the `is_flat` method and add the checks in the setters but I just thought I'd avoid code duplication that way.

Reply:
how about a private method ```_```